### PR TITLE
Spring cleaning + fix claims closed+reset reserves bug

### PR DESCRIPTION
**Pull request by palmfjord**

### CODE REVIEW

File: src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java [hunk 1]
```diff
@@ -81,82 +80,78 @@ public void on(BackofficeClaimCreatedEvent e, @Timestamp Instant timestamp) {
...
+    ClaimReportHistoryEntity claimHistoryEntry = copyLatestClaimHistoryEntity(e.getClaimID(), timestamp);
 
-    updatedClaimHistoryEntry.setReserved(BigDecimal.valueOf(e.amount));
-    updatedClaimHistoryEntry.setCurrency(SEK);
-    claimReportHistoryRepository.save(updatedClaimHistoryEntry);
+    claimHistoryEntry.setReserved(BigDecimal.valueOf(e.amount));
+    claimHistoryEntry.setCurrency(SEK);
+
+    final Instant timeOfKnowledgeAdjustedForRaceCondition = claimHistoryEntry.getTimeOfKnowledge().plusMillis(1);
+    claimHistoryEntry.setTimeOfKnowledge(timeOfKnowledgeAdjustedForRaceCondition); // Until we meet again, missing-reserves-bug
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java]: goodbye bug, you will be missed!

File: src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java [hunk 2]
```diff
@@ -182,31 +177,32 @@ public void on(AutomaticPaymentInitiatedEvent e, @Timestamp Instant timestamp) {
...
+  @ResetHandler
+  public void onReset() {
+    log.warn("Deleting all claim report history entities");
+    claimReportHistoryRepository.deleteAll();
+  }
+
+  private ClaimReportHistoryEntity copyLatestClaimHistoryEntity(String claimId, Instant timeOfKnowledge) {
     List<ClaimReportHistoryEntity> listOfClaims = claimReportHistoryRepository.findByClaimId(claimId);
 
+    Instant timeOfKnowledgeWithRaceConditionWiggleRoom = timeOfKnowledge.plusMillis(500);
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java]: is this still needed ❓

Comment (palmfjord) [reply src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java]: Slightly unsure, i can rerun the generation without it and see what happens

Comment (johantjellden) [reply src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java]: What was the racecondition about?

Comment (palmfjord) [reply src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java]: This happens at the same millisecond sometimes 

https://github.com/HedvigInsurance/claims-service/blob/a2df74cde81506d3a129708c35f6827301578203/src/main/java/com/hedvig/claims/aggregates/ClaimsAggregate.java#L77-L93

Causing these guys to fail

https://github.com/HedvigInsurance/claims-service/blob/39b181be3ae9eef4c752a438ee51015f1f9fc12a/src/main/java/com/hedvig/claims/services/ReportGenerationServiceImpl.java#L76

https://github.com/HedvigInsurance/claims-service/blob/ae2016d0a5290a13856c50c92afd8ab709acd0a9/src/main/java/com/hedvig/claims/query/ClaimReportHistoryEventListener.java#L197

**Pull request merged**